src/main/java/com/minecolonies/coremod/entity/pathfinding/MinecoloniesAdvancedPathNavigate.java

```diff
@@ -330,6 +325,13 @@ else if (world.getBlockState(ourEntity.getPosition()).getBlock() == Blocks.GRASS
 		}
     }
 
+	public Block findBlockTypeUnderEntity(Entity parEntity)
+{
+    int blockX = MathHelper.floor(parEntity.posX);
+    int blockY = MathHelper.floor(parEntity.posY-0.2D);
+    int blockZ = MathHelper.floor(parEntity.posZ);
+    return world.getBlockState(new BlockPos(blockX, blockY, blockZ)).getBlock() ;
```

Comment:
Maybe round would be better for x and z ?

Reply:
Possibly yes.